src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,141 @@
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int capacity) {
+        array = (T[]) new Object[capacity];
+    }
+
     @Override
     public void add(T value) {
+        if (size == array.length) {
+            grow();
+        }
 
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "The index is out of bounds. Size of the ArrayList is "
+                            + size + ". Your index is " + index
+            );
+        }
 
+        if (size == array.length) {
+            grow();
+        }
+
+        for (int i = size; i > index; i--) {
+            array[i] = array[i - 1];
+        }
+
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index > size - 1) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "The index is out of bounds. Size of the ArrayList is "
+                            + size + ". Your index is " + index
+            );
+        }
+
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
+        if (index < 0 || index > size - 1) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "The index is out of bounds. Size of the ArrayList is "
+                            + size + ". Your index is " + index
+            );
+        }
 
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index < 0 || index > size - 1) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "The index is out of bounds. Size of the ArrayList is "
+                            + size + ". Your index is " + index
+            );
+        }
+
+        final T removedElement = array[index];
+
+        for (int i = index; i < size - 1; i++) {
+            array[i] = array[i + 1];
+        }
+
+        array[size - 1] = null;
+        size--;
+
+        return removedElement;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (element == null && array[i] == null
+                    || element != null && element.equals(array[i])) {
+                return remove(i);
+            }
+        }
+
+        throw new NoSuchElementException();
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
```

Comment:
Don't begin class or method implementation with an empty line. Remove the empty line at the start of the 'isEmpty' method.